First-time build of .NET glue from source generates .NET runtime errors because config file does not yet exist

### Godot version

4.1dev (a7d0e18a31)

### System information

Windows 11, .NET SDK 6.0.407/7.0.202

### Issue description

When following the instructions on the [Compiling with .NET guide](https://docs.godotengine.org/en/latest/contributing/development/compiling/compiling_with_dotnet.html), the very first attempt at the ["Generate the glue"](https://docs.godotengine.org/en/latest/contributing/development/compiling/compiling_with_dotnet.html#generate-the-glue) step, with a clean `bin` directory, generates errors about being unable to find the .NET runtime, even with x64 .NET SDKs installed. These errors are apparently due to a missing configuration file, and do not actually seem to prevent successfully completing a .NET build. However, they could certainly appear to be showstopping issues to new users.

After invoking:
```
> scons p=windows target=editor module_mono_enabled=yes vsproj=yes [dev_build=yes]
> bin\godot.windows.editor[.dev].x86_64.mono --headless --generate-mono-glue modules\mono\glue
```

The following command-line errors and pop-up error dialog are shown (although the Godot binary proceeds to generate a large number of `.cs` files afterward):
```
ERROR: hostfxr_initialize_for_runtime_config failed with code: -2147450733
   at: (modules\mono\mono_gd\gd_mono.cpp:201)
ERROR: .NET: Failed to load compatible .NET runtime
   at: (modules\mono\mono_gd\gd_mono.cpp:278)
ERROR: Parameter "godot_plugins_initialize" is null.
   at: GDMono::initialize (modules\mono\mono_gd\gd_mono.cpp:411)
```
<img width="294" alt="Error dialog" src="https://user-images.githubusercontent.com/23439518/226438453-8b33739c-444b-498e-84af-57eaea2e0660.png">

With a dev build, debugging shows the failure is in an attempt to load the configuration file `bin\GodotSharp\Api\Debug\GodotPlugins.runtimeconfig.json`. This file does not yet exist, but _will be generated as part of the next step in the instructions_, ["Building the managed libraries"](https://docs.godotengine.org/en/latest/contributing/development/compiling/compiling_with_dotnet.html#building-the-managed-libraries).

After building the managed libraries, once the JSON config file is in place, the .NET errors do not appear in subsequent invocations of the Godot binary for generating the glue code, and the resulting dev build appears to successfully open and run a .NET Godot project.

I've prepared an edit to the Compiling with .NET documentation page, which I could set up a PR for -- it places a note on the glue step warning users about these errors. However, I'm unsure if that's the "right" solution or if some other change to the code or the docs is more appropriate (e.g., if the configuration file could be generated as part of the glue step, instead of the managed-libraries step).

### Steps to reproduce

- Clone a clean copy of the Godot source on Windows 11 with the .NET SDK installed, or delete all contents of the `bin` directory in the Godot source folder in an existing copy
- Run `scons` with `module_mono_enabled=yes`
- Run `bin\godot.windows.edit.x86_64.mono --headless --generate-mono-glue modules\mono\glue`
- Errors appear
- To resolve, run `python modules\mono\build_scripts\build_assemblies.py --godot-output-dir=bin` and try the `--generate-mono-glue` step again

### Minimal reproduction project

N/A

Running `build_assemblies.py` no longer works for me:

```
❯ ./modules/mono/build_scripts/build_assemblies.py --godot-output-dir=./bin

Running MSBuild:  /usr/bin/dotnet msbuild /home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp.sln /restore /t:Build /p:Configuration=Debug /p:NoWarn=1591
MSBuild version 17.4.1+fedecea9d for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj(142,3): error MSB4019: The imported project "/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GeneratedIncludes.props" was not found. Confirm that the expression in the Import declaration "Generated/GeneratedIncludes.props" is correct, and that the file exists on disk.
/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharpEditor/GodotSharpEditor.csproj(43,3): error MSB4019: The imported project "/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharpEditor/Generated/GeneratedIncludes.props" was not found. Confirm that the expression in the Import declaration "Generated/GeneratedIncludes.props" is correct, and that the file exists on disk.
/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj(142,3): error MSB4019: The imported project "/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GeneratedIncludes.props" was not found. Confirm that the expression in the Import declaration "Generated/GeneratedIncludes.props" is correct, and that the file exists on disk.
  Godot.SourceGenerators.Internal -> /home/geo/dev/godot/modules/mono/glue/GodotSharp/Godot.SourceGenerators.Internal/bin/Debug/netstandard2.0/Godot.SourceGenerators.Internal.dll
```

I cannot build the mono version of the editor anymore.
I'm unsure why? Can somebody explain how to build the mono version from a clean slate? I don't think the documentation is up-to-date.

--- a/modules/mono/csharp_script.cpp
+++ b/modules/mono/csharp_script.cpp
@@ -93,6 +93,12 @@ String CSharpLanguage::get_extension() const {
 }
 
 void CSharpLanguage::init() {
+#ifdef TOOLS_ENABLED
+	if (OS::get_singleton()->get_cmdline_args().find("--generate-mono-glue")) {
+		print_verbose(".NET: Skipping runtime initialization because glue generation is enabled.");
+		return;
+	}
+#endif
 #ifdef DEBUG_METHODS_ENABLED
 	if (OS::get_singleton()->get_cmdline_args().find("--class-db-json")) {
 		class_db_api_to_json("user://class_db_api.json", ClassDB::API_CORE);

